File: qqbot-protocol/src/main/java/com/mieai/qqbot/protocol/openapi/QqContentModels.java
package com.mieai.qqbot.protocol.openapi;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.mieai.qqbot.protocol.openapi.QqGuildModels.Member;
import com.mieai.qqbot.protocol.openapi.QqGuildModels.User;
import java.util.List;

/** Wire models for QQ reactions, announcements, pins, schedules, forums, and audio. */
public final class QqContentModels {
    private QqContentModels() {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record ReactionUsers(
            List<User> users,
            String cookie,
            @JsonProperty("is_end") Boolean end) {}

    public record RecommendChannel(
            @JsonProperty("channel_id") String channelId,
            String introduce) {}

    public record GuildAnnouncementRequest(
            @JsonProperty("channel_id") String channelId,
            @JsonProperty("message_id") String messageId,
            @JsonProperty("announces_type") Integer announcementType,
            @JsonProperty("recommend_channels") List<RecommendChannel> recommendChannels) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record Announcement(
            @JsonProperty("guild_id") String guildId,
            @JsonProperty("channel_id") String channelId,
            @JsonProperty("message_id") String messageId,
            @JsonProperty("announces_type") Integer announcementType,
            @JsonProperty("recommend_channels") List<RecommendChannel> recommendChannels) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record Pins(
            @JsonProperty("guild_id") String guildId,
            @JsonProperty("channel_id") String channelId,
            @JsonProperty("message_ids") List<String> messageIds) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record Schedule(
            String id,
            String name,
            String description,
            @JsonProperty("start_timestamp") String startTimestamp,
            @JsonProperty("end_timestamp") String endTimestamp,
            Member creator,
            @JsonProperty("jump_channel_id") String jumpChannelId,
            @JsonProperty("remind_type") String remindType) {}

    public record ScheduleInput(
            String name,
            String description,
            @JsonProperty("start_timestamp") String startTimestamp,
            @JsonProperty("end_timestamp") String endTimestamp,
            @JsonProperty("jump_channel_id") String jumpChannelId,
            @JsonProperty("remind_type") String remindType) {}

    public record ScheduleRequest(ScheduleInput schedule) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record ForumThreadInfo(
            @JsonProperty("thread_id") String threadId,
            String title,
            String content,
            @JsonProperty("date_time") String dateTime) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record ForumThread(
            @JsonProperty("guild_id") String guildId,
            @JsonProperty("channel_id") String channelId,
            @JsonProperty("author_id") String authorId,
            @JsonProperty("thread_info") ForumThreadInfo threadInfo) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record ForumThreads(
            List<ForumThread> threads,
            @JsonProperty("is_finish") Integer finished) {}

    public record ForumThreadRequest(String title, String content, Integer format) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record ForumThreadCreateResult(
            @JsonProperty("task_id") String taskId,
            @JsonProperty("create_time") String createTime) {}

    public record AudioControl(
            @JsonProperty("audio_url") String audioUrl,
            String text,
            Integer status) {}
}

File: qqbot-protocol/src/main/kotlin/com/mieai/qqbot/protocol/openapi/QqContentModels.kt
package com.mieai.qqbot.protocol.openapi

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import com.mieai.qqbot.protocol.openapi.QqGuildModels.Member
import com.mieai.qqbot.protocol.openapi.QqGuildModels.User

/** Wire models for QQ reactions, announcements, pins, schedules, forums, and audio. */
object QqContentModels {
    @JsonIgnoreProperties(ignoreUnknown = true)
    data class ReactionUsers(
        val users: List<User>?,
        val cookie: String?,
        @JsonProperty("is_end") val end: Boolean?,
    )

    data class RecommendChannel(
        @JsonProperty("channel_id") val channelId: String?,
        val introduce: String?,
    )

    data class GuildAnnouncementRequest(
        @JsonProperty("channel_id") val channelId: String?,
        @JsonProperty("message_id") val messageId: String?,
        @JsonProperty("announces_type") val announcementType: Int?,
        @JsonProperty("recommend_channels") val recommendChannels: List<RecommendChannel>?,
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Announcement(
        @JsonProperty("guild_id") val guildId: String?,
        @JsonProperty("channel_id") val channelId: String?,
        @JsonProperty("message_id") val messageId: String?,
        @JsonProperty("announces_type") val announcementType: Int?,
        @JsonProperty("recommend_channels") val recommendChannels: List<RecommendChannel>?,
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Pins(
        @JsonProperty("guild_id") val guildId: String?,
        @JsonProperty("channel_id") val channelId: String?,
        @JsonProperty("message_ids") val messageIds: List<String>?,
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Schedule(
        val id: String?,
        val name: String?,
        val description: String?,
        @JsonProperty("start_timestamp") val startTimestamp: String?,
        @JsonProperty("end_timestamp") val endTimestamp: String?,
        val creator: Member?,
        @JsonProperty("jump_channel_id") val jumpChannelId: String?,
        @JsonProperty("remind_type") val remindType: String?,
    )

    data class ScheduleInput(
        val name: String?,
        val description: String?,
        @JsonProperty("start_timestamp") val startTimestamp: String?,
        @JsonProperty("end_timestamp") val endTimestamp: String?,
        @JsonProperty("jump_channel_id") val jumpChannelId: String?,
        @JsonProperty("remind_type") val remindType: String?,
    )

    data class ScheduleRequest(val schedule: ScheduleInput?)

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class ForumThreadInfo(
        @JsonProperty("thread_id") val threadId: String?,
        val title: String?,
        val content: String?,
        @JsonProperty("date_time") val dateTime: String?,
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class ForumThread(
        @JsonProperty("guild_id") val guildId: String?,
        @JsonProperty("channel_id") val channelId: String?,
        @JsonProperty("author_id") val authorId: String?,
        @JsonProperty("thread_info") val threadInfo: ForumThreadInfo?,
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class ForumThreads(
        val threads: List<ForumThread>?,
        @JsonProperty("is_finish") val finished: Int?,
    )

    data class ForumThreadRequest(val title: String?, val content: String?, val format: Int?)

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class ForumThreadCreateResult(
        @JsonProperty("task_id") val taskId: String?,
        @JsonProperty("create_time") val createTime: String?,
    )

    data class AudioControl(
        @JsonProperty("audio_url") val audioUrl: String?,
        val text: String?,
        val status: Int?,
    )
}
